packages/x-charts-pro/src/ChartsBrushOverlay/ChartsBrushOverlay.tsx

```diff
@@ -0,0 +1,141 @@
+'use client';
+import * as React from 'react';
+import { useStore, useSelector, selectorChartDrawingArea } from '@mui/x-charts/internals';
+import clsx from 'clsx';
+import { useTheme } from '@mui/material/styles';
+import {
+  selectorChartBrushConfig,
+  selectorChartBrushCurrentX,
+  selectorChartBrushCurrentY,
+  selectorChartBrushStartX,
+  selectorChartBrushStartY,
+} from '../internals/plugins/useChartProZoom/useChartProZoom.selectors';
+import type { UseChartProZoomSignature } from '../plugins';
+import { brushOverlayClasses } from './ChartsBrushOverlay.classes';
+
+function BrushLine(props: React.SVGProps<SVGLineElement>) {
+  return (
+    <line
+      className={brushOverlayClasses.line}
+      strokeWidth={2}
+      strokeDasharray={'4 2'}
+      pointerEvents={'none'}
+      {...props}
+    />
+  );
+}
+
+function BrushRect(props: React.SVGProps<SVGRectElement>) {
+  return (
+    <rect
+      className={brushOverlayClasses.rect}
+      strokeWidth={1}
+      fillOpacity={0.2}
+      pointerEvents={'none'}
+      {...props}
+    />
+  );
+}
+
+export interface ChartsBrushOverlayProps {}
+
+/**
+ * Component that renders visual feedback during brush zoom interaction
+ */
+export function ChartsBrushOverlay(_: ChartsBrushOverlayProps) {
+  const store = useStore<[UseChartProZoomSignature]>();
+  const drawingArea = useSelector(store, selectorChartDrawingArea);
+
+  const theme = useTheme();
+
+  const brushStartX = useSelector(store, selectorChartBrushStartX);
+  const brushStartY = useSelector(store, selectorChartBrushStartY);
+  const brushCurrentX = useSelector(store, selectorChartBrushCurrentX);
+  const brushCurrentY = useSelector(store, selectorChartBrushCurrentY);
+  const brushConfig = useSelector(store, selectorChartBrushConfig);
+  if (!brushStartX || !brushStartY || !brushCurrentX || !brushCurrentY) {
+    return null;
+  }
+
+  const { left, top, width, height } = drawingArea;
+
+  // Clamp coordinates to drawing area
+  const clampX = (x: number) => Math.max(left, Math.min(left + width, x));
+  const clampY = (y: number) => Math.max(top, Math.min(top + height, y));
+
+  const startX = clampX(brushStartX);
+  const startY = clampY(brushStartY);
+  const currentX = clampX(brushCurrentX);
+  const currentY = clampY(brushCurrentY);
+  const rectColor = theme.palette.mode === 'light' ? 'grey' : 'white';
+  const strokeColor = theme.palette.mode === 'light' ? '#000000' : '#ffffff';
+
+  // For scatter charts, show only the rectangle without guide lines
+  if (brushConfig === 'xy') {
+    const rectWidth = currentX - startX;
+    const rectHeight = currentY - startY;
```

Comment:
We can apply `Math.abs()` here so we don't need to do it twice later 

```suggestion
    const rectWidth = Math.abs(currentX - startX);
    const rectHeight = Math.abs(currentY - startY);
```